polkadot/consensus/src/dynamic_inclusion.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Dynamic inclusion threshold over time.
+
+use std::time::{Duration, Instant};
+
+fn duration_to_micros(duration: &Duration) -> u64 {
+	duration.as_secs() * 1_000_000 + (duration.subsec_nanos() / 1000) as u64
+}
+
+/// Dynamic inclusion threshold over time.
+///
+/// The acceptable proportion of parachains which must have parachain candidates
+/// reduces over time (eventually going to zero).
+#[derive(Debug, Clone)]
+pub struct DynamicInclusion {
+	start: Instant,
+	y: i64,
```

Comment:
there is a comment a bit further down describing the meaning of the slope/intercept. Since `m` is always negative, `m*x` is as well, and since we want `y + m*x` it has to be signed. Changing to use unsigned arithmetic and clarifying the comment.